Hide notices on Business Services tab in marketplace

KokkieH opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->
If someone has a WCCOM extension installed, we show a notice to connect their store to WCCOM in the in-app marketplace. 

If their store is connected, we show a notice to install/activate the WooCommerce.com Update Manager plugin as needed.

We don't want these notices to appear on the Business Services tab, as this type of product requires neither a connection to WCCOM nor do they need to update.

This PR also fixes a bug where the WUM notice appeared multiple times on the search results tab, once for every product type, and includes a small refactor to the Search Results tab to simplify the code a little.

Closes https://github.com/Automattic/woocommerce.com/issues/20493

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure your WCCOM environment is running on the latest version of the `feature/business-services-tab` branch.
2. Make sure you have a few products on the Business Services category
3. Check out this branch
4. In `plugins/woocommerce-admin/client/marketplace/components/constants.ts`, edit the `MARKETPLACE_HOST` const to be `https://woocommerce.test`
5. Go to `/plugins/woocommerce` and run `pnpm --filter='@woocommerce/plugin-woocommerce' build`
6. Visit [WooCommerce.com](https://woocommerce.com/my-account/downloads/) and download any plugin. Install it on your local site and activate.
7. Visit the [in-app marketplace](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions). Provided your site is not connected to WCCOM, you should see a notice prompting you to connect your site
![Extensions_‹_WooCommerce_‹_woocommerce](https://github.com/woocommerce/woocommerce/assets/11873759/62c79c3c-4f69-437b-9719-954d5933e7f8)
8. Check all the tabs (including Search results by searching for something and also My Subscriptions) and confirm the notice appears on all tabs except Business Services
9. Click the user icon top-left and connect your store to your WCCOM account.
10. Return to the in-app marketplace. Now you should see a notice prompting you to install the WooCommerce.com Update Manager plugin:
![Extensions_‹_WooCommerce_‹_woocommerce](https://github.com/woocommerce/woocommerce/assets/11873759/402d9554-b9e5-415a-adbb-03acb7e97b5b)
11. Again check all the tabs and confirm the notice does not appear on the Business Services tab
12. Search for a term that will return multiple product types ("business" works for me). Confirm that multiple sections load on the search results tab, based on the results, but that the WUM notice only appears once at the top, not before every section.
13. If search results don't include all three product types, confirm the tab doesn't show empty components for the types not included in results (e.g. if results don't include business service products, the Search Results tab shouldn't show a Business Services section)
14. See what you can break ;)

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

KokkieH committed:
Do not display Woo Update Manager notice on Business Services tab - in-app marketplace

KokkieH committed:
Render <PluginInstallNotice/> inside <Content/> component instead of <Products/>

- Ensures notice is not rendered multiple times on Search tab

- Also remove <PluginInstallNotice/> from Discover and My Subscriptions components to avoid duplication

- Hide <ConnectNotice/> on Business Services tab as well

KokkieH committed:
Refactor Search Results code to simplify and improve readability

- Reduce number of conditions we check and just render search result components selectively in final return instead

mcliwanow commented on plugins/woocommerce-admin/client/marketplace/components/search-results/search-results.tsx:
```diff
@@ -163,57 +163,24 @@ export default function SearchResults( props: SearchResultProps ): JSX.Element {
 			} );
 		}
 
-		if ( hasExtensions && hasThemes && ! hasBusinessServices ) {
```
How about removing lines 142-165 too?

KokkieH replied:
Done in 3e89baa

KokkieH committed:
Further refactoring to Search Results component

- Assign component props selectively, so we do not need to check so many conditions

mcliwanow commented on plugins/woocommerce-admin/client/marketplace/components/content/content.tsx:
```diff
@@ -185,7 +186,12 @@ export default function Content(): JSX.Element {
 			<Promotions />
 			<InstallNewProductModal products={ products } />
-			<ConnectNotice />
+			{ selectedTab !== 'business-services' && (
+				<>
+					<ConnectNotice />
+					<PluginInstallNotice />
+				</>
+			) }
```
If we wanted to remove `<ConnectNotice />` from My Subscriptions screen (there's already information about missing connection there), the code that I tested is:
 
```suggestion
			{ selectedTab !== 'business-services' && <PluginInstallNotice /> }
			{ selectedTab !== 'business-services' &&
				selectedTab !== 'my-subscriptions' && <ConnectNotice /> }
```

TBD though if we want to change it.

KokkieH replied:
Done in e01b89a after getting confirmation from @hesahesa

mcliwanow reviewed (approved):
Thanks for the changes, @KokkieH. The code is much simpler now and the unnecessary notices are not displayed in the Business Services tab 🙌

mcliwanow committed:
In-App Marketplace - update search placeholder when there are no business services

If there are no business services to be displayed, we shouldn't include 'and business services' in the placeholder as it can be misleading

KokkieH committed:
Do not display ConnectNotice on My Subscriptions tab either
